homeassistant/components/graphite/__init__.py

```diff
@@ -100,12 +112,16 @@ def event_listener(self, event):
 
     def _send_to_graphite(self, data):
         """Send data to Graphite."""
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        sock.settimeout(10)
-        sock.connect((self._host, self._port))
-        sock.sendall(data.encode("ascii"))
-        sock.send(b"\n")
-        sock.close()
+        if self._protocol == PROTOCOL_TCP:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(10)
+            sock.connect((self._host, self._port))
+            sock.sendall(data.encode("ascii"))
+            sock.send(b"\n")
+            sock.close()
+        else:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.sendto(data.encode("ascii") + b"\n", (self._host, self._port))
```

Comment:
All in all, this PR looks good, however, this might need to be extracted into a separate Python package hosted on Pypi. I'll ask for a second opinion on that.